### PR TITLE
JDK-8272192: Shortcut String equality checks by checking equality of the value array Export

### PR DESCRIPTION
**Pull request by q2q-2q2**
less string comparison in equals and compareTo when
- same string object
- strings are deduplicated

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Error
&nbsp;⚠️ OCA signatory status must be verified

### Integration blocker
&nbsp;⚠️ Title mismatch between PR and JBS for issue [JDK-8272192](https://bugs.openjdk.org/browse/JDK-8272192)

### Issue
 * [JDK-8272192](https://bugs.openjdk.org/browse/JDK-8272192): Shortcut String equality checks by checking equality of the value array ⚠️ Title mismatch between PR and JBS.

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12363/head:pull/12363` \
`$ git checkout pull/12363`

Update a local copy of the PR: \
`$ git checkout pull/12363` \
`$ git pull https://git.openjdk.org/jdk.git pull/12363/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12363`

View PR using the GUI difftool: \
`$ git pr show -t 12363`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12363.diff">https://git.openjdk.org/jdk/pull/12363.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/String.java [hunk 1]
```diff
@@ -1874,6 +1874,9 @@ private boolean nonSyncContentEquals(AbstractStringBuilder sb) {
         byte v2[] = sb.getValue();
         byte coder = coder();
         if (coder == sb.getCoder()) {
+            if (v1 == v2) {
```
Comment (turbanoff) [inline src/java.base/share/classes/java/lang/String.java:1877]: How is this possible? Array inside a StringBuilder is modifiable, while array inside String is immutable.

Comment (q2q-2q2) [reply src/java.base/share/classes/java/lang/String.java:1877]: You are right. Thank you.

**Comment by q2q-2q2** [comment]
My OCA has been processed in accordance with the mail I received on 06/29/2021. Is it possible for it to expire?

**Comment by q2q-2q2** [comment]
It is still oca labelled and no communications about it. Can anyone help?

**Comment by q2q-2q2** [comment]
Pre-submit tests - linux-x86 / test - Test (tier1) fails with message:
Unable to find an artifact with the name: bundles-linux-x86-debug

**Pull request closed without merging**